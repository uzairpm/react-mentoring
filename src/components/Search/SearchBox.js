import React, {Component} from 'react';

import Button from '../Common/Button';

const jumbotron = {
    padding: '1rem 1rem',
    marginBottom: 0
};
const searchInput = {
    borderBottom: '2px solid #fa6c85'
};
const widerButton = {
    padding: '0.25rem 2rem'
};

export default class SearchBox extends Component {
    constructor(props) {
        super(props);
        this._handleKeyPress = this._handleKeyPress.bind(this);
    }
    componentDidMount() {
        if (location.hash.indexOf('/search/') !== -1) {
            const searchQuery = location.hash.substr(9);
            if (searchQuery.length > 0) {
                this.props.titleClick();
                this.props.valChange(searchQuery);
                this.props.searchClick();
            }
        }
    }
    _handleKeyPress(e) {
        if (e.key === 'Enter') {
            this.props.searchClick();
        }
    }
    render() {
        return (
            <div className="jumbotron" style={jumbotron}>
                <p className="lead">FIND YOUR MOVIE</p>
                <div className="input-group mb-3">
                    <input type="text" 
                        className="form-control rounded-0 searchInput"
                        style={searchInput}
                        onChange={(e) => this.props.valChange(e.target.value) }
                        value={ this.props.value }
                        onKeyPress={this._handleKeyPress} />
                </div>
                <label className="mr-3">SEARCH BY</label>
                <Button label="TITLE"
                    clickHandler={this.props.titleClick}
                    style={widerButton}
                    className={"widerButton btn btn-sm mr-3 " + (this.props.titleActive ? 'btn-dark' : 'btn-outline-dark')}/>
                <Button label="GENRE"
                    style={widerButton}
                    clickHandler={this.props.genreClick}
                    className={"widerButton btn btn-sm " + (this.props.titleActive ? 'btn-outline-dark' : 'btn-dark')}/>
                <Button label="SEARCH" 
                    style={widerButton}
                    className="widerButton btn btn-primary btn-sm float-right" 
                    type="submit" clickHandler={this.props.searchClick}/>
            </div>
        );
    }
}
