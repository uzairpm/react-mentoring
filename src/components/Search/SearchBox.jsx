import React, {Component} from 'react';
import styled from 'styled-components';

import Button from '../Common/Button';

const Jumbotron = styled.div`
    padding: 1rem 1rem;
    margin-bottom: 0;
`;
const SearchInput = styled.input`
    border-bottom: 2px solid #fa6c85;
`;
const WiderButton = styled(Button)`
    padding: 0.25rem 2rem;
`;

export default class SearchBox extends Component {
    constructor(props) {
        super(props);
        this._handleKeyPress = this._handleKeyPress.bind(this);
    }
    componentDidMount() {
        if (location.hash.indexOf('/search/') !== -1) {
            const searchQuery = location.hash.substr(9);
            if (searchQuery.length > 0) {
                this.props.titleClick();
                this.props.valChange(searchQuery);
                this.props.searchClick();
            }
        }
    }
    _handleKeyPress(e) {
        if (e.key === 'Enter') {
            this.props.searchClick();
        }
    }
    render() {
        return (
            <Jumbotron className="jumbotron">
                <p className="lead">FIND YOUR MOVIE</p>
                <div className="input-group mb-3">
                    <SearchInput type="text" 
                        className="form-control rounded-0 searchInput"
                        onChange={(e) => this.props.valChange(e.target.value) }
                        value={ this.props.value }
                        onKeyPress={this._handleKeyPress} />
                </div>
                <label className="mr-3">SEARCH BY</label>
                <WiderButton label="TITLE"
                    clickHandler={this.props.titleClick}
                    className={"btn btn-sm mr-3 " + (this.props.titleActive ? 'btn-dark' : 'btn-outline-dark')}/>
                <WiderButton label="GENRE"
                    clickHandler={this.props.genreClick}
                    className={"btn btn-sm " + (this.props.titleActive ? 'btn-outline-dark' : 'btn-dark')}/>
                <WiderButton label="SEARCH" 
                    className="btn btn-primary btn-sm float-right" 
                    type="submit" clickHandler={this.props.searchClick}/>
            </Jumbotron>
        );
    }
}
