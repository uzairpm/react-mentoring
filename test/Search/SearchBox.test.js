import React from 'react';

import SearchBox from '../../src/components/Search/SearchBox';

describe('Check Component creation', () => {
    test('SearchBox component', () => {
        const onChangeText = jest.fn();
        const onClickSearch = jest.fn();
        const element = shallow(<SearchBox titleActive={true}
            value={'matrix'}
            valChange={onChangeText}
            searchClick={onClickSearch}/>);
        // console.log(element.debug());
        expect(element).toMatchSnapshot();
    });
});
