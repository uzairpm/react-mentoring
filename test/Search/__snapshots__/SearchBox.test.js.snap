// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchBox Component Snapshot test with custom values for props 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <SearchBox
    searchClick={
      [MockFunction] {
        "calls": Array [
          Array [],
        ],
      }
    }
    titleActive={false}
    valChange={[MockFunction]}
    value="mat"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <p
          className="lead"
        >
          FIND YOUR MOVIE
        </p>,
        <div
          className="input-group mb-3"
        >
          <input
            className="form-control rounded-0 searchInput"
            onChange={[Function]}
            onKeyPress={[Function]}
            style={
              Object {
                "borderBottom": "2px solid #fa6c85",
              }
            }
            type="text"
            value="mat"
          />
        </div>,
        <label
          className="mr-3"
        >
          SEARCH BY
        </label>,
        <Button
          className="widerButton btn btn-sm mr-3 btn-outline-dark"
          clickHandler={undefined}
          label="TITLE"
          style={
            Object {
              "padding": "0.25rem 2rem",
            }
          }
        />,
        <Button
          className="widerButton btn btn-sm btn-dark"
          clickHandler={undefined}
          label="GENRE"
          style={
            Object {
              "padding": "0.25rem 2rem",
            }
          }
        />,
        <Button
          className="widerButton btn btn-primary btn-sm float-right"
          clickHandler={
            [MockFunction] {
              "calls": Array [
                Array [],
              ],
            }
          }
          label="SEARCH"
          style={
            Object {
              "padding": "0.25rem 2rem",
            }
          }
          type="submit"
        />,
      ],
      "className": "jumbotron",
      "style": Object {
        "marginBottom": 0,
        "padding": "1rem 1rem",
      },
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "FIND YOUR MOVIE",
          "className": "lead",
        },
        "ref": null,
        "rendered": "FIND YOUR MOVIE",
        "type": "p",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <input
            className="form-control rounded-0 searchInput"
            onChange={[Function]}
            onKeyPress={[Function]}
            style={
              Object {
                "borderBottom": "2px solid #fa6c85",
              }
            }
            type="text"
            value="mat"
          />,
          "className": "input-group mb-3",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "className": "form-control rounded-0 searchInput",
            "onChange": [Function],
            "onKeyPress": [Function],
            "style": Object {
              "borderBottom": "2px solid #fa6c85",
            },
            "type": "text",
            "value": "mat",
          },
          "ref": null,
          "rendered": null,
          "type": "input",
        },
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "SEARCH BY",
          "className": "mr-3",
        },
        "ref": null,
        "rendered": "SEARCH BY",
        "type": "label",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "className": "widerButton btn btn-sm mr-3 btn-outline-dark",
          "clickHandler": undefined,
          "label": "TITLE",
          "style": Object {
            "padding": "0.25rem 2rem",
          },
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "className": "widerButton btn btn-sm btn-dark",
          "clickHandler": undefined,
          "label": "GENRE",
          "style": Object {
            "padding": "0.25rem 2rem",
          },
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "className": "widerButton btn btn-primary btn-sm float-right",
          "clickHandler": [MockFunction] {
            "calls": Array [
              Array [],
            ],
          },
          "label": "SEARCH",
          "style": Object {
            "padding": "0.25rem 2rem",
          },
          "type": "submit",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <p
            className="lead"
          >
            FIND YOUR MOVIE
          </p>,
          <div
            className="input-group mb-3"
          >
            <input
              className="form-control rounded-0 searchInput"
              onChange={[Function]}
              onKeyPress={[Function]}
              style={
                Object {
                  "borderBottom": "2px solid #fa6c85",
                }
              }
              type="text"
              value="mat"
            />
          </div>,
          <label
            className="mr-3"
          >
            SEARCH BY
          </label>,
          <Button
            className="widerButton btn btn-sm mr-3 btn-outline-dark"
            clickHandler={undefined}
            label="TITLE"
            style={
              Object {
                "padding": "0.25rem 2rem",
              }
            }
          />,
          <Button
            className="widerButton btn btn-sm btn-dark"
            clickHandler={undefined}
            label="GENRE"
            style={
              Object {
                "padding": "0.25rem 2rem",
              }
            }
          />,
          <Button
            className="widerButton btn btn-primary btn-sm float-right"
            clickHandler={
              [MockFunction] {
                "calls": Array [
                  Array [],
                ],
              }
            }
            label="SEARCH"
            style={
              Object {
                "padding": "0.25rem 2rem",
              }
            }
            type="submit"
          />,
        ],
        "className": "jumbotron",
        "style": Object {
          "marginBottom": 0,
          "padding": "1rem 1rem",
        },
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "FIND YOUR MOVIE",
            "className": "lead",
          },
          "ref": null,
          "rendered": "FIND YOUR MOVIE",
          "type": "p",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <input
              className="form-control rounded-0 searchInput"
              onChange={[Function]}
              onKeyPress={[Function]}
              style={
                Object {
                  "borderBottom": "2px solid #fa6c85",
                }
              }
              type="text"
              value="mat"
            />,
            "className": "input-group mb-3",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "className": "form-control rounded-0 searchInput",
              "onChange": [Function],
              "onKeyPress": [Function],
              "style": Object {
                "borderBottom": "2px solid #fa6c85",
              },
              "type": "text",
              "value": "mat",
            },
            "ref": null,
            "rendered": null,
            "type": "input",
          },
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "SEARCH BY",
            "className": "mr-3",
          },
          "ref": null,
          "rendered": "SEARCH BY",
          "type": "label",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "className": "widerButton btn btn-sm mr-3 btn-outline-dark",
            "clickHandler": undefined,
            "label": "TITLE",
            "style": Object {
              "padding": "0.25rem 2rem",
            },
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "className": "widerButton btn btn-sm btn-dark",
            "clickHandler": undefined,
            "label": "GENRE",
            "style": Object {
              "padding": "0.25rem 2rem",
            },
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "className": "widerButton btn btn-primary btn-sm float-right",
            "clickHandler": [MockFunction] {
              "calls": Array [
                Array [],
              ],
            },
            "label": "SEARCH",
            "style": Object {
              "padding": "0.25rem 2rem",
            },
            "type": "submit",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`SearchBox Component Snapshot test with default props 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <SearchBox
    searchClick={[MockFunction]}
    titleActive={true}
    valChange={[MockFunction]}
    value="matrix"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <p
          className="lead"
        >
          FIND YOUR MOVIE
        </p>,
        <div
          className="input-group mb-3"
        >
          <input
            className="form-control rounded-0 searchInput"
            onChange={[Function]}
            onKeyPress={[Function]}
            style={
              Object {
                "borderBottom": "2px solid #fa6c85",
              }
            }
            type="text"
            value="matrix"
          />
        </div>,
        <label
          className="mr-3"
        >
          SEARCH BY
        </label>,
        <Button
          className="widerButton btn btn-sm mr-3 btn-dark"
          clickHandler={undefined}
          label="TITLE"
          style={
            Object {
              "padding": "0.25rem 2rem",
            }
          }
        />,
        <Button
          className="widerButton btn btn-sm btn-outline-dark"
          clickHandler={undefined}
          label="GENRE"
          style={
            Object {
              "padding": "0.25rem 2rem",
            }
          }
        />,
        <Button
          className="widerButton btn btn-primary btn-sm float-right"
          clickHandler={[MockFunction]}
          label="SEARCH"
          style={
            Object {
              "padding": "0.25rem 2rem",
            }
          }
          type="submit"
        />,
      ],
      "className": "jumbotron",
      "style": Object {
        "marginBottom": 0,
        "padding": "1rem 1rem",
      },
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "FIND YOUR MOVIE",
          "className": "lead",
        },
        "ref": null,
        "rendered": "FIND YOUR MOVIE",
        "type": "p",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <input
            className="form-control rounded-0 searchInput"
            onChange={[Function]}
            onKeyPress={[Function]}
            style={
              Object {
                "borderBottom": "2px solid #fa6c85",
              }
            }
            type="text"
            value="matrix"
          />,
          "className": "input-group mb-3",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "className": "form-control rounded-0 searchInput",
            "onChange": [Function],
            "onKeyPress": [Function],
            "style": Object {
              "borderBottom": "2px solid #fa6c85",
            },
            "type": "text",
            "value": "matrix",
          },
          "ref": null,
          "rendered": null,
          "type": "input",
        },
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "SEARCH BY",
          "className": "mr-3",
        },
        "ref": null,
        "rendered": "SEARCH BY",
        "type": "label",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "className": "widerButton btn btn-sm mr-3 btn-dark",
          "clickHandler": undefined,
          "label": "TITLE",
          "style": Object {
            "padding": "0.25rem 2rem",
          },
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "className": "widerButton btn btn-sm btn-outline-dark",
          "clickHandler": undefined,
          "label": "GENRE",
          "style": Object {
            "padding": "0.25rem 2rem",
          },
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "className": "widerButton btn btn-primary btn-sm float-right",
          "clickHandler": [MockFunction],
          "label": "SEARCH",
          "style": Object {
            "padding": "0.25rem 2rem",
          },
          "type": "submit",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <p
            className="lead"
          >
            FIND YOUR MOVIE
          </p>,
          <div
            className="input-group mb-3"
          >
            <input
              className="form-control rounded-0 searchInput"
              onChange={[Function]}
              onKeyPress={[Function]}
              style={
                Object {
                  "borderBottom": "2px solid #fa6c85",
                }
              }
              type="text"
              value="matrix"
            />
          </div>,
          <label
            className="mr-3"
          >
            SEARCH BY
          </label>,
          <Button
            className="widerButton btn btn-sm mr-3 btn-dark"
            clickHandler={undefined}
            label="TITLE"
            style={
              Object {
                "padding": "0.25rem 2rem",
              }
            }
          />,
          <Button
            className="widerButton btn btn-sm btn-outline-dark"
            clickHandler={undefined}
            label="GENRE"
            style={
              Object {
                "padding": "0.25rem 2rem",
              }
            }
          />,
          <Button
            className="widerButton btn btn-primary btn-sm float-right"
            clickHandler={[MockFunction]}
            label="SEARCH"
            style={
              Object {
                "padding": "0.25rem 2rem",
              }
            }
            type="submit"
          />,
        ],
        "className": "jumbotron",
        "style": Object {
          "marginBottom": 0,
          "padding": "1rem 1rem",
        },
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "FIND YOUR MOVIE",
            "className": "lead",
          },
          "ref": null,
          "rendered": "FIND YOUR MOVIE",
          "type": "p",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <input
              className="form-control rounded-0 searchInput"
              onChange={[Function]}
              onKeyPress={[Function]}
              style={
                Object {
                  "borderBottom": "2px solid #fa6c85",
                }
              }
              type="text"
              value="matrix"
            />,
            "className": "input-group mb-3",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "className": "form-control rounded-0 searchInput",
              "onChange": [Function],
              "onKeyPress": [Function],
              "style": Object {
                "borderBottom": "2px solid #fa6c85",
              },
              "type": "text",
              "value": "matrix",
            },
            "ref": null,
            "rendered": null,
            "type": "input",
          },
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "SEARCH BY",
            "className": "mr-3",
          },
          "ref": null,
          "rendered": "SEARCH BY",
          "type": "label",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "className": "widerButton btn btn-sm mr-3 btn-dark",
            "clickHandler": undefined,
            "label": "TITLE",
            "style": Object {
              "padding": "0.25rem 2rem",
            },
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "className": "widerButton btn btn-sm btn-outline-dark",
            "clickHandler": undefined,
            "label": "GENRE",
            "style": Object {
              "padding": "0.25rem 2rem",
            },
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "className": "widerButton btn btn-primary btn-sm float-right",
            "clickHandler": [MockFunction],
            "label": "SEARCH",
            "style": Object {
              "padding": "0.25rem 2rem",
            },
            "type": "submit",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
